xpartition.py

```diff
@@ -353,14 +353,27 @@ def indexers(self, ranks: int, rank: int, dims: Sequence[Hashable]) -> Region:
             dask_indexers = meta_array.blocks.indexers(**meta_indexers)
             return self._obj.blocks.indexers(**dask_indexers)
 
-    def write(self, store: str, ranks: int, dims: Sequence[Hashable], rank: int):
-        _write_partition_dataarray(self._obj, store, ranks, dims, rank)
+    def write(
+        self,
+        store: str,
+        ranks: int,
+        dims: Sequence[Hashable],
+        rank: int,
+        collect_variable_writes: bool = False,
+    ):
+        self.to_dataset().partition.write(
+            store, ranks, dims, rank, collect_variable_writes
+        )
```

Comment:
This doesn't feel as clean as before, but I think this may be cleanest way we can handle writing DataArrays that contain chunked coordinates.